Sources/Private/Experimental/ExperimentalAnimationLayer.swift

```diff
@@ -138,7 +133,50 @@ final class ExperimentalAnimationLayer: CALayer {
 
 }
 
-// MARK: RootAnimationLayer
+// MARK: - CALayer + setupLayerHierarchy
+
+extension CALayer {
+  /// Sets up an `AnimationLayer` / `CALayer` hierarchy in this layer,
+  /// using the given list of layers.
+  func setupLayerHierarchy(
+    for layers: [LayerModel],
+    context: LayerContext)
+  {
+    var layersByIndex = [Int: CALayer]()
+
+    // First, we build the `AnimationLayer` / `CALayer` for each `LayerModel`
+    for layerModel in layers.reversed() {
+      guard let layer = layerModel.makeAnimationLayer(context: context) else {
+        continue
+      }
+
+      layersByIndex[layerModel.index] = layer
+    }
+
+    // Then we add each `AnimationLayer` to the layer hierarchy
+    for layerModel in layers.reversed() {
```

Comment:
We may not need to use `reversed()` here, will double-check.

> And might be useful to document why we do two passes, I assume it's because we can't guarantee that the nodes come topologically sorted?

Yep, there are no ordering guarantees. Most of the layers in existing sample files have an ordering that requires two passes.